Demo/Demo/FeatureViewControllers/FeatureBaseViewController.swift

```diff
@@ -93,12 +93,9 @@ class FeatureBaseViewController: UIViewController, ASWebAuthenticationPresentati
     // MARK: - ASWebAuthenticationPresentationContextProviding conformance
 
     func presentationAnchor(for session: ASWebAuthenticationSession) -> ASPresentationAnchor {
-        UIApplication
-            .shared
-            .connectedScenes
-            .flatMap { ($0 as? UIWindowScene)?.windows ?? [] }
-            .first { $0.isKeyWindow }
-        ?? ASPresentationAnchor()
+        let firstScene = UIApplication.shared.connectedScenes.first as? UIWindowScene
+        let window = firstScene?.windows.first { $0.isKeyWindow }
+        return window ?? ASPresentationAnchor()
```

Comment:
Nice! This is certainly a lot cleaner. Left a comment on how we can also resolve the main thread warning!